### PR TITLE
Refine Feature Selector

### PR DESCRIPTION
**Pull request by xuehui1991**
1. Update the doc of Overview.
2. Add benchmark.

### CODE REVIEW

**Comment by QuanluZhang** [comment]
looks great!

File: docs/en_US/FeatureEngineering/Overview.md [hunk 1]
```diff
@@ -1,3 +1,238 @@
...
+        Set the parameters of this estimator.
+        """
+        for param in params:
+            if hasattr(self, param):
+                setattr(self, param, params[param])
+        return self
+
+```
+
+**1. Inherit the SelectorMixin Class and its Function**
```
Comment (chicm-ms) [inline docs/en_US/FeatureEngineering/Overview.md]: should be `**2` here ?

File: docs/en_US/FeatureEngineering/Overview.md [hunk 2]
```diff
@@ -1,3 +1,238 @@
...
+
+For now, we support the following feature selector:
+- [GradientFeatureSelector](./GradientFeatureSelector.md)
+- [GBDTSelector](./GBDTSelector.md)
+
+
+# How to use?
+
+```python
+from nni.feature_engineering.xxx_selector import XXXSelector
```
Comment (chicm-ms) [inline docs/en_US/FeatureEngineering/Overview.md]: Suggest to use a real selector such as `GradientFeatureSelector` instead of `XXXSelector`

File: docs/en_US/FeatureEngineering/Overview.md [hunk 3]
```diff
@@ -1,3 +1,238 @@
...
+            regression). Which shape is [n_samples].
+        """
+        self.X = X
+        self.y = y
+        ...
+    
+    def get_selected_features(self):
+        """
+        Fit the training data to FeatureSelector
+
```
Comment (chicm-ms) [inline docs/en_US/FeatureEngineering/Overview.md]: doc string needs to be updated.

File: docs/en_US/FeatureEngineering/Overview.md [hunk 4]
```diff
@@ -1,3 +1,238 @@
...
+        Returns
+        -------
+        list :
+                Return the index of imprtant feature.
```
Comment (chicm-ms) [inline docs/en_US/FeatureEngineering/Overview.md]: 1. typo: important
2.  what is 'important feature'?  need to be clarified.

File: docs/en_US/FeatureEngineering/Overview.md [hunk 5]
```diff
@@ -1,3 +1,238 @@
...
+        ...
+        return self.selected_features_
+
+    ...
+```
+
+**3. Integrate with Sklearn**
+
+`sklearn.pipeline.Pipeline` can connect models in series, such as feature selector, normalization, and classification/regression to form a typical machine learning problem workflow. 
+The fllowing step could help us to better integrate with sklearn, which means we could treat the customized feature selector as a mudule of pipeline.
```
Comment (chicm-ms) [inline docs/en_US/FeatureEngineering/Overview.md]: typo: mudule

File: docs/en_US/FeatureEngineering/Overview.md (hunk 1 above)
Comment (xuehui1991) [reply docs/en_US/FeatureEngineering/Overview.md]: yes

File: docs/en_US/FeatureEngineering/Overview.md [hunk 6]
```diff
@@ -1,3 +1,257 @@
...
+```python
+from nni.feature_engineering.gradient_selector import GradientFeatureSelector
+# from nni.feature_engineering.gbdt_selector import GBDTSelector
+
+# load data
+...
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
+
+# initlize a selector
+fgs = GradientFeatureSelector(...)
```
Comment (QuanluZhang) [inline docs/en_US/FeatureEngineering/Overview.md]: gfs?

File: docs/en_US/FeatureEngineering/Overview.md [hunk 7]
```diff
@@ -1,3 +1,257 @@
...
+`Baseline` means without any feature selection, we directly pass the data to LogisticRegression. For this benchmark, we only use 10% data from the train as test data.
+
+| Dataset | Baseline | GradientFeatureSelector | TreeBasedClassifier | #Train | #Feature | 
+| ----------- | ------ | ------ | ------- | ------- | -------- |
+| colon-cancer | 0.7547 | 0.7368 | 0.7223 | 62 | 2,000 |
+| gisette | 0.9725 | 0.89416 | 0.9792 | 6,000 | 5,000 |
+| avazu | 0.8834 | N/A | N/A | 40,428,967 | 1,000,000 |
+| rcv1 | 0.9644 | 0.7333 | 0.9615 | 20,242 | 47,236 |
+| news20.binary | 0.9208 | 0.6870 | 0.9070 | 19,996 | 1,355,191 |
+| real-sim | 0.9681 | 0.7969 | 0.9591 | 72,309 | 20,958 |
```
Comment (QuanluZhang) [inline docs/en_US/FeatureEngineering/Overview.md]: no number of gbdtselector?
